main.py

```diff
@@ -1398,7 +1402,7 @@ def run(self, edit):
                     "uri": filename_to_uri(self.view.file_name())
                 },
                 "options": {
-                    "tabSize": 4,
+                    "tabSize": 4,  # TODO: Fetch these from the project settings / global settings
```

Comment:
Just saw this merged. The tab size can be easily read by `"tabSize": self.view.settings().get("tab_size", 4)` This includes all available sources: view specific setting, project setting, user setting, default.

Reply:
Yeah. It’s also hardcoded in the format document request, so I felt that copying that behavior was the best course of action. We should fix it in another PR (so fix both format doc and format selection options).